File: 3254. Find the Power of K-Size Subarrays I/find_the_power_of_k_size_subarrays_i.py
class Solution:
    def resultsArray(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        res = []

        i = 0
        while i <= n - k:
            is_valid = True
            for j in range(i, i + k - 1):
                if nums[j+1] - nums[j] != 1:
                    is_valid = False
                    break

            if is_valid:
                res.append(nums[i+k-1])
                i += 1
            else:
                res.extend([-1] * (min(n - k, j) + 1 - i))
                i = j + 1

        return res

File: 3254. Find the Power of K-Size Subarrays I/find_the_power_of_k_size_subarrays_i.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k) {
        if (k == 1) {
            return nums;
        }

        vector<int> powers(nums.size() - k + 1, -1);
        int length = 1;
        for (int i = 0; i < nums.size() - 1; ++i) {
            if (nums[i] + 1 == nums[i+1]) {
                ++length;
            } else {
                length = 1;
            }

            if (length >= k) {
                powers[i-k+2] = nums[i+1];
            }
        }

        return powers;
    }
};